framework/tlvf/src/src/CmduMessageTx.cpp

```diff
@@ -81,11 +83,23 @@ bool CmduMessageTx::finalize(bool swap_needed)
     if (!m_cmdu_header)
         return false; 
 
-    if (m_cmdu_header->message_type() == eMessageType::VENDOR_SPECIFIC_MESSAGE &&
-        getClassCount() > 1) {
-        tlvVendorSpecific *tlv = dynamic_cast<tlvVendorSpecific *>(getClass(0).get());
-        if (tlv != nullptr) {
-            tlv->length() = getMessageLength() - (kCmduHeaderLength + kTlvHeaderLength);
+    // Update vendor specific length which is seperated to 3 classes
+    for (auto class_it = m_class_vector.begin(); class_it != m_class_vector.end(); class_it++) {
+        // Type is the first byte in the tlv buffer
+        uint8_t tlv_type = *(*class_it)->getStartBuffPtr();
+        if (tlv_type == uint8_t(eTlvType::TLV_VENDOR_SPECIFIC)) {
+            uint16_t* tlv_length = reinterpret_cast<uint16_t *>
+                ((class_it->get())->getStartBuffPtr() + sizeof(uint8_t));
+            // tlv vendor specific length already include OUI and beerocks header length
+            // need to add only beerocks message length.
+            // Classes order: +0 = tlvVendorSpecific, +1 = beerocks_header, +2 = beerocks_message
+            if (std::next(class_it, 1) == m_class_vector.end() || 
```

Comment:
There's a redundant space at the end of the line, and the same two lines below.

For some reason, clang-format skips over the framework/tlvf/src directory...